src/core/ddsc/src/dds_reader.c

```diff
@@ -622,6 +622,19 @@ static dds_entity_t dds_create_reader_int (dds_entity_t participant_or_subscribe
   if (tp->m_ktopic->qos)
```

Comment:
This is a good reason why the entity name shouldn't be set on the QoS. The
```C
  if (sub->m_entity.m_qos)
    ddsi_xqos_mergein_missing (rqos, sub->m_entity.m_qos, ~(uint64_t)0);
```
above (which GitHub doesn't allow me attach on comment on, silly GitHub) I think means it would copy the entity name from the subscriber.

Using `~QP_ENTITY_NAME` instead of `~(uint64_t)0` for the `ddsi_xqos_mergein_missing` means it doesn't copy the entity name. That should solve it.